### PR TITLE
#6 Update state depends on the userRole

### PR DESCRIPTION
**Pull request by Whooolia**

### CODE REVIEW

File: src/component/form/form.js [hunk 1]
```diff
@@ -11,12 +11,15 @@ class Form extends Component {
   // hardcoded state for testing purposes
   state = {
     email: "jjj@jjj.com",
-    pin: "2437"
+    pin: "2437",
+    userRole: "driver"
```
Comment (Jbarget) [inline src/component/form/form.js]: lets put the userRoles into constants. Constants are typically defined in a constants file and are all uppercase:

in `src/constants/userRoles.js`:

```
export const DRIVER = 'DRIVER'
export const SUPPLIER = 'SUPPLIER
```

then, anytime you need to refer to a driver you import the constant and use it, not sure what the other user roles are but put them into constants too :)

File: src/component/form/form.js [hunk 2]
```diff
@@ -11,12 +11,15 @@ class Form extends Component {
...
   };
   handleChange = event => {
     const target = event.target;
+    // const value = target.type === 'radio' ? target.id : target.value
```
Comment (Jbarget) [inline src/component/form/form.js]: can we get rid of this?

File: src/component/form/form.js [hunk 3]
```diff
@@ -11,12 +11,15 @@ class Form extends Component {
   // hardcoded state for testing purposes
   state = {
     email: "jjj@jjj.com",
-    pin: "2437"
+    pin: "2437",
+    userRole: "driver"
   };
   handleChange = event => {
```
Comment (Jbarget) [inline src/component/form/form.js]:
```
handleChange = { target } => {
	this.setState({...})
}
```

**Review by Jbarget — CHANGES_REQUESTED** [review]

File: src/component/form/form.js (hunk 3 above)
Comment (Oliversw) [reply src/component/form/form.js]: Ohh I love this!

File: src/component/form/form.js (hunk 1 above)
Comment (Oliversw) [reply src/component/form/form.js]: Could you please explain more why we would benefit from using this? We actually only have the two user roles, and I don't think that we need to refer to them in too many places.

Comment (Jbarget) [reply src/component/form/form.js]: this is something that is so easy to implement at the beginning but if you want to implement further down the line its a lot harder. So its not just about whats needed right now

There's a phrase in coding called "single source of truth". As a general rule always always always have a single source of truth for something. So in this case wherever you refer to the user being a `DRIVER` or a `SUPPLIER` it always comes from one place

Comment (Oliversw) [reply src/component/form/form.js]: Oh okay! So in that case every constant that we create throughout the app would be contained in that file?

**Comment by Whooolia** [comment]
The last commit is about authentication. Now It's fully functional authentication for drivers and suppliers! :+1: Updates state with the data we got back from Airtable or it gives false if the data is not matched to Airtable

File: src/component/form/form.js (hunk 1 above)
Comment (Jbarget) [reply src/component/form/form.js]: "would be contained in that file" do you mean folder?

`src/constants/...`

Comment (charlielafosse) [reply src/component/form/form.js]: Hey I think we've correctly implemented this now! Just pushed up

**Pull request merged**